docs/securing_installation.md

```diff
@@ -0,0 +1,108 @@
+
+
+### The Tiller gRPC Endpoint and TLS
+
+In the default installation the gRPC endpoint that Tiller offers is available inside the cluster (not external to the cluster) without authentication configuration applied. Without applying authentication, any process in the cluster can use the gRPC endpoint to perform operations inside the cluster. In a local or secured private cluster, this enables rapid usage and is normal. (From outside the cluster, Helm uses the normal authentication that Kubernetes supports to reach a gRPC endpoint.)
+
+Shared and production clusters -- for the most part -- should use Helm 2.3.0 at a minimum and configure TLS for the Tiller gRPC endpoints to ensure that within the cluster usage of the gRPC endpoint is only for the properly authenticated identity. Doing so enables any number of Tiller instances to be deployed in any number of namespaces and yet no unauthenticated usage of any gRPC endpoint is possible. Finally, usage of Helm with the `--tiller-tls-verify` option to enforce TLS in any communication with gRPC endpoints.
+
+For the proper steps to configure Tiller and use Helm properly with TLS configured, see [Using SSL between Helm and Tiller](tiller_ssl.md).
+
+When Helm clients are connecting from outside of the cluster, the security between the Helm client and the API server is managed by Kubernetes itself. You may want to ensure that this link is secure. Note that if you are using the TLS configuration recommended above, not even the Kubernetes API server has access to the unencrypted messages between the client and Tiller.
+
+### Tiller's Release Information
+
+For historical reasons, Tiller stores its release information in ConfigMaps. We suggest changing the default to Secrets.
+
+Secrets are the Kubernetes accepted mechanism for saving configuration data that is considered sensitive. While secrets don't themselves offer many protections, Kubernetes cluster management software often treats them differently than other objects. Thus, we suggest using secrets to store releases.
+
+Enabling this feature currently requires setting the `--storage=secret` flag in the tiller-deploy deployment. This entails directly modifying the deployment or using `helm init --override=...`, as no helm init flag is currently available to do this for you. For more information, see [Using --override](install.md#using---override).
+
+### Thinking about Charts
+
+Because of the relative longevity of Helm, the Helm chart ecosystem evolved without the immediate concern for cluster-wide control, and especially in the developer space this makes complete sense. However, charts are a kind of package that not only installs containers you may or may not have validated yourself, but it may also install into more than one namespace. 
```

Comment:
I don't have it handy right now (I want to say `--dry-run --debug`), but there is a way to have the helm client dump the yaml that will be installed.  Paranoid users should use this to get a concrete idea of what is going to happen before they install a chart.  Might be good to note that here.

Reply:
`helm template`.

I disagree. I would actually link to how users can [verify the integrity and origin of a package](https://github.com/kubernetes/helm/blob/master/docs/provenance.md) instead. Most users won't be informed enough to make the call that the output from `helm template` is safe or unsafe. Installing and verifying trusted packages would be the better approach for paranoid sysadmins.

Reply:
Up to you what you want to tell users.  

My advice to users is to look deeper whenever pulling from public places.  This includes helm charts or public docker images.  At heptio we got a dev cluster owned by this issue: https://github.com/kubernetes/charts/issues/1341.

Reply:
@jbeda's specific case is a good example to use to decide how to handle the issue. Without clear documentation about applying a more secure configuration, Jenkins was arbitrarily assuming that a LoadBalancer was "OK". Which it most certainly was not. 

Jenkins fixed the chart by not installing that way by default. That is the proper fix for the chart.

@bacongobbler I read the doc about provenance to establish the identity of the chart you're pulling in. However, this says nothing about whether the verified chart installs Jenkins's JNLP master port as a LoadBalancer by default -- though its provenance could be demonstrated. In this case, using Jenkins's earlier chart would in fact have been prevented had the reader of this document thought hey, I should probably review that chart for how it configures the ports. Or if Tiller in a particular internet-exposed cluster had been configured to be unable to create a LoadBalancer. 

I do not see how this says anything about helm or tiller's security _per se_, but I do think that it's in line to emphasize how installing other people's software without due diligence can set up configurations that you would not have thought proper had you looked -- depending on your RBAC/security configuration.

It's this issue that I want to illuminate in this case. It's not a vulnerability in a cluster that has been tightened down in the way that we describe here. But many clusters will continue to have for _very good reasons_ RBAC scopes that are more "cluster admin" in breadth, and those should quite rightly require a bit (if not a damned sight!) more diligence about what the charts DO and what images they call IN than merely verifying _their provenance_. 

Do I misunderstand, or does that make my current position more clear?